x-pack/plugins/spaces/public/views/management/components/delete_spaces_button.js

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { Component, Fragment } from 'react';
+import PropTypes from 'prop-types';
+import { ConfirmDeleteModal } from './confirm_delete_modal';
+import {
+  EuiButton
+} from '@elastic/eui';
+import { toastNotifications } from 'ui/notify';
+
+export class DeleteSpacesButton extends Component {
+  state = {
+    showConfirmModal: false
+  };
+
+  render() {
+    const numSpaces = this.props.spaces.length;
+
+    const buttonText = numSpaces > 1
+      ? `Delete ${numSpaces} spaces`
+      : `Delete space`;
+
+    return (
+      <Fragment>
+        <EuiButton color={'danger'} onClick={this.onDeleteClick}>
+          {buttonText}
+        </EuiButton>
+        {this.getConfirmDeleteModal()}
+      </Fragment>
+    );
+  }
+
+  onDeleteClick = () => {
+    this.setState({
+      showConfirmModal: true
+    });
+  };
+
+  getConfirmDeleteModal = () => {
+    if (!this.state.showConfirmModal) {
+      return null;
+    }
+
+    return (
+      <ConfirmDeleteModal
+        spaces={this.props.spaces}
+        onCancel={() => {
+          this.setState({
+            showConfirmModal: false
+          });
+        }}
+        onConfirm={this.deleteSpaces}
+      />
+    );
+  };
+
+  deleteSpaces = () => {
+    const {
+      httpAgent,
+      chrome,
+      spaces,
+      $q
+    } = this.props;
+
+    console.log(this.props, spaces);
+
+    const deleteOperations = spaces.map(space => {
+      return httpAgent.delete(
+        chrome.addBasePath(`/api/spaces/v1/spaces/${encodeURIComponent(space.id)}`)
+      );
+    });
+
+    $q.all(deleteOperations)
```

Comment:
I keep forgetting that `Promise.all` exists on native Promises...I'll change to use that. I brought `$q` in to prevent pulling in another dependency, but it's a non-issue. I'll fix this too.